src/core/agent/agent.ts

```diff
@@ -238,32 +240,65 @@ class Agent {
         throw new Error(Agent.KERIA_BOOTED_ALREADY_BUT_CANNOT_CONNECT);
       }
       await this.saveAgentUrls(agentUrls);
-      Agent.isOnline = true;
-      this.agentServicesProps.signifyClient = this.signifyClient;
-      this.agentServicesProps.eventService.emit<KeriaStatusChangedEvent>({
-        type: KeriaStatusEventTypes.KeriaStatusChanged,
-        payload: {
-          isOnline: Agent.isOnline,
-        },
-      });
+      this.activeSignifyClient();
     }
   }
+  async recovery(seedPhrase: string[], connectUrl: string): Promise<void> {
+    try {
+      const mnemonic = seedPhrase.join(" ");
+      const bran = Buffer.from(mnemonicToEntropy(mnemonic), "hex")
+        .toString("utf-8")
+        .replace(/\0/g, "");
 
-  private async saveAgentUrls(agentUrls: AgentUrls): Promise<void> {
-    await this.basicStorageService.save({
-      id: MiscRecordId.KERIA_CONNECT_URL,
-      content: {
-        url: agentUrls.url,
-      },
+      this.signifyClient = new SignifyClient(connectUrl, bran, Tier.low);
+
+      await this.signifyClient.connect();
+    } catch (error) {
+      if (error instanceof Error) {
+        if (error.message === "Invalid mnemonic")
+          throw new Error(Agent.KERIA_SEEDPHRASE_INVALID);
+        if (error.message.includes("agent does not exist for controller"))
+          throw new Error(Agent.KERA_BOOT_URL_NOT_FOUND);
+        throw Error(Agent.KERA_RECOVERY_FAILED);
```

Comment:
This buries the error and doesn't allow us to debug what the issue was. Instead, we can just do `throw error;`